### ISSUE TITLE
Calling writeJSON on one-shot getar dump causes segfault

### ISSUE THREAD

**Issue by bdice** [glotzerlab/hoomd-blue#354]
This minimal reproduction script causes a segfault when attempting to write to a getar dump created in one-shot mode. This should create a user-readable Python exception. I read a bit of the source code but I was not able to determine when the file is actually closed from the Python or C++ interfaces. In any case, the Python-level object should be able to ask the C++ side if its underlying archive is still readable/writable or not and perform that check before doing I/O operations triggered on the Python side.

```python
import hoomd
from hoomd import hpmc

hoomd.context.initialize()
system=hoomd.init.create_lattice(hoomd.lattice.sc(a=2), n=2)
mc = hpmc.integrate.convex_polyhedron(seed=42);
cube_vertices = [( 0.5, 0.5, 0.5), (-0.5, 0.5, 0.5),
                 ( 0.5,-0.5, 0.5), (-0.5,-0.5, 0.5),
                 ( 0.5, 0.5,-0.5), (-0.5, 0.5,-0.5),
                 ( 0.5,-0.5,-0.5), (-0.5,-0.5,-0.5)]
mc.shape_param.set('A', vertices=cube_vertices)

filename = 'output.zip'
getar = hoomd.dump.getar('oneshot.zip', '1',
                         static=['viz_static'],
                         dynamic={'viz_aniso_dynamic': 1})
getar.writeJSON('type_shapes.json', mc.get_type_shapes(), dynamic=False)
hoomd.run(10)
```

**Comment by joaander** [comment]
@bdice Can we get a fix in for v2.6.1?

**Comment by bdice** [comment]
I don't have time to fix this right now. This could be a hackathon project. I vote to close with `wontfix` unless this issue recurs with other users.
